converter/internal/staticconvert/internal/build/builder.go

```diff
@@ -0,0 +1,84 @@
+package build
+
+import (
+	"fmt"
+
+	"github.com/grafana/agent/component/discovery"
+	"github.com/grafana/agent/converter/diag"
+	"github.com/grafana/agent/converter/internal/prometheusconvert"
+	"github.com/grafana/agent/pkg/config"
+	"github.com/grafana/agent/pkg/integrations/apache_http"
+	int_config "github.com/grafana/agent/pkg/integrations/config"
+	"github.com/grafana/agent/pkg/integrations/node_exporter"
+	"github.com/grafana/agent/pkg/river/token/builder"
+	"github.com/prometheus/common/model"
+	prom_config "github.com/prometheus/prometheus/config"
+)
+
+type IntegrationsV1ConfigBuilder struct {
+	f         *builder.File
+	diags     *diag.Diagnostics
+	cfg       *config.Config
+	globalCtx *GlobalContext
+}
+
+func NewIntegrationsV1ConfigBuilder(f *builder.File, diags *diag.Diagnostics, cfg *config.Config, globalCtx *GlobalContext) *IntegrationsV1ConfigBuilder {
+	return &IntegrationsV1ConfigBuilder{
+		f:         f,
+		diags:     diags,
+		cfg:       cfg,
+		globalCtx: globalCtx,
+	}
+}
+
+func (b *IntegrationsV1ConfigBuilder) AppendIntegrations() {
+	for _, integration := range b.cfg.Integrations.ConfigV1.Integrations {
+		if !integration.Common.Enabled {
+			continue
+		}
+
+		var exports discovery.Exports
+		switch itg := integration.Config.(type) {
+		case *apache_http.Config:
+			exports = b.AppendApacheExporter(itg, &integration.Common)
+		case *node_exporter.Config:
+			exports = b.AppendNodeExporter(itg, &integration.Common)
```

Comment:
Should we have a clause for unsupported integration?
a new one can be added, without adding support in the config converters

Reply:
That gets handled in the separate validate.go functionality with a similar case statement for unsupported integrations. This will be up for discussion and refactor if we want to do it inline once the dust settles on the first 3 converters.